mod-infra/src/main/scala/at/forsyte/apalache/infra/Executor.scala

```diff
@@ -0,0 +1,39 @@
+package at.forsyte.apalache.infra
+
+import at.forsyte.apalache.infra.passes.{Pass, PassChainExecutor, ToolModule, WriteablePassOptions}
+import com.google.inject.Guice
+
+/**
+ * This Executor abstracts the dependency injection and execution logic required for executing a PassChainExecutor
+ *
+ * @param toolModule
+ *   The tool module that specifies the sequence of passes
+ * @author
+ *   Shon Feder
+ */
+case class Executor(val toolModule: ToolModule) {
+  private val injector = Guice.createInjector(toolModule)
+
+  /** Exposes mutable access to options configuring the behavior of the [passChainExecutor] */
+  val passOptions = injector.getInstance(classOf[WriteablePassOptions])
+
+  private val exceptionAdapter = injector.getInstance(classOf[ExceptionAdapter])
```

Comment:
I'm not sure what the benefit of that would be given our current architecture. The point of this class at the moment is that all of these values are functions of the `toolModule` supplied. So I'm not sure how or why we'd make it so that these values that depend on the one parameter we already supply could be constructed and supplied separately.

Reply:
> So I'm not sure how or why we'd make it so that these values that depend on the one parameter we already supply could be constructed and supplied separately.

We should assume that other objects than the ones (transitively) created by `Executor` will need dependency injection in the future. Before, we were constructing the injector in `Tool`, which I believe is the right place to [ensure a single instance](https://github.com/google/guice/wiki/InjectingTheInjector).
The way to ensure this is to create the injector outside of `Executor`, statically declare `Executor`'s dependencies, and have `passOptions` and `exceptionAdapter` injected.

We only depend on `toolModule` here for a hack: To use `injector` as a [service locator](https://www.baeldung.com/java-service-locator-pattern) to inject a variable number of `Pass`-typed arguments. I suppose there is no way to statically declare variadic arguments as dependencies in Guice?

Even so, I think it's cleaner to create the injector outside, use dependency injection where we can, and pass the (partially redundant) `toolModule` for working around the variadic dependency.